### PR TITLE
Fix error-prone warnings in test code

### PR DESCRIPTION
**Pull request by alsutton**
### Overview

When I was making some other changes I noticed we were getting 100 warnings in the test code, so I spent a bit of time cleaning them up.

### Proposed Changes

The changes follow the error prone recommendations. Most of the changes are updates to use a more modern API or language features.

### CODE REVIEW

File: robolectric/src/test/java/org/robolectric/shadows/ShadowDisplayTest.java [hunk 1]
```diff
@@ -171,8 +171,8 @@ public void setDisplayHdrCapabilities_shouldReturnHdrCapabilities() {
   }
 
   @Test
-  @Config(minSdk = M)
-  public void setDisplayHdrCapabilities_shouldThrowUnSupportedOperationException() {
+  @Config(maxSdk = M)
```
Comment (hoisie) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowDisplayTest.java:174]: Can the changes to ShadowDisplayTest be put in a different PR to keep history clean? Seems like it is a little more than just error prone cleanup.

Comment (alsutton) [reply robolectric/src/test/java/org/robolectric/shadows/ShadowDisplayTest.java]: Sure, done.

File: robolectric/src/test/java/org/robolectric/shadows/ShadowHandlerTest.java [hunk 2]
```diff
@@ -74,7 +73,8 @@ public void testDefaultConstructorUsesDefaultLooper() throws Exception {
   }
 
   private static Looper newLooper(boolean canQuit) {
-    return ReflectionHelpers.callConstructor(Looper.class, from(boolean.class, canQuit));
+    return ReflectionHelpers.callConstructor(
+        Looper.class, ReflectionHelpers.ClassParameter.from(boolean.class, canQuit));
```
Comment (hoisie) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowHandlerTest.java:77]: nit: can you change all of these to just `ClassParameter.from`? So that would require importing `import org.robolectric.util.ReflectionHelpers.ClassParameter`. That's the most common pattern in Robolectric.

**Comment by hoisie** [comment]
Thanks for doing all of this! Only the 'ClassParameter.from' nit. Would it be possible to rebase on origin/master and drop the last 2 commits? Thanks!

File: robolectric/src/test/java/org/robolectric/shadows/ShadowHandlerTest.java (hunk 2 above)
Comment (alsutton) [reply robolectric/src/test/java/org/robolectric/shadows/ShadowHandlerTest.java]: Done.

**Comment by alsutton** [comment]
No problem. Done.
